docs/contrib/DeveloperGuide.md

```diff
@@ -318,6 +318,16 @@ clang-tidy -p ./build source_file.cpp
 clang-format -i -style=file:.clang-format source_file.cpp
 ```
 
+To run `clang-tidy` on all files, you can use the following script that runs
+`clang-tidy` in parallel and also caches the results, so subsequent runs
+only have to operate on changed files.
```

Comment:
Could this be accomplished through cmake?  It sounds very similar to what is already does in terms of parallelism and updating.

Reply:
I don't know, but cmake is a somewhat black box to me. 
Maybe it can be done to run `clang-tidy` in parallel, but I suspect the sophisticated persistent caching can not be easily achieved.